tests/utils/test_client.py

```diff
@@ -54,3 +54,13 @@ def test_client_enforce_healthy():
     assert my_client.protocol == CONFIG["my_instance_name"]["protocol"]
     assert my_client.base_path == "/api/versioned/v1/"
     assert my_client.auth == UsernamePasswordAuth("admin", os.environ["TAMR_TOOLBOX_PASSWORD"],)
+
+
+@mock_api()
+def test_get_with_connection_retry():
+    # To ensure timeout error is raised to test correct ConnectionError is caught
+    my_client = utils.client.create(**CONFIG["my_other_instance"])
+    with pytest.raises(TimeoutError):
+        utils.client.get_with_connection_retry(
+            my_client, "/api/service/health", timeout_seconds=10, sleep_seconds=1
+        )
```

Comment:
I believe that this test will ensure that an uncaught `ConnectionError` is being avoided, but I worry that there might be some way a `TimeoutError` could be raised that isn't the intended goal here.

Your code solution writes a log message with specific content, so I think it would be better to also check that this is occurring.  See [this test](https://github.com/Datatamer/tamr-toolbox/blob/b51e7079b5103c96c0c6a1fd629d490426dd7763/tests/utils/test_logger.py#L80-L112) for how to have code run in a test log to a temp file, and then how to check that file's contents.  Ignore the part about running a script, that's not necessary here.

Reply:
Ok, will look into it.